doc/kci_testsuite.md

```diff
@@ -0,0 +1,300 @@
+
+8. Add appropriate test entry with LAVA job definition for `btrfs-verify-cli.sh`. For example,
+
+```
+run:
+    steps:
+        - lava-test-case test-btrfs-cli-001 --shell ls /usr/bin/btrfs-verify-cli.sh
+
+```
+
+If you like read more about LAVA job definition, please check [here](https://docs.lavasoftware.org/lava/writing-tests.html)
+
+
+### How to add tests to LAVA
+
+#### Prerequisites
+
+- Rootfs image is built 
+- Rootfs image boots on a platform
+- Tests can be run
+
+If you're facing issues with tha above you may want to go through the section "How to add test suite to a rootfs image"
+
+#### Adding new test plan to LAVA
+
+1. Use the rootfs image to boot the device and produce a LAVA job definition for the tests.
+Upload rootfs image to kernelci-backend (storage)
+
+```
+kci_rootfs upload --rootfs-dir /path/to/rootf/image/ --upload-path server/path --api http://api.kernelci --token KCI_API_TOKEN
+```
+
+Add rootfs definition to `test-configs.yaml`
+
+```yaml
+  debian_buster-v4l2_ramdisk:
+    type: debian
+    ramdisk: 'buster-v4l2/20200626.0/{arch}/rootfs.cpio.gz'
+```
+
+`ramdisk` path must be relative to storage server URL.
+
+Detailed description of the test configuration can be found [here](https://github.com/kernelci/kernelci-doc/wiki/Test-configurations)
+
+2. Add LAVA job definition template
+
+Create appropriate directory structure under `templates` directory.
+
+```
+mkdir templates/v4l2-compliance/
+```
+
+Create test plan definition
+
+```
+cat templates/v4l2-compliance/v4l2-compliance.jinja2
+```
+
+```yaml
+- test:
+    timeout:
+      minutes: 10
+    definitions:
+    - repository:
+        metadata:
+          format: Lava-Test Test Definition 1.0
+          name: {{ plan }}
+          description: "v4l2 test plan"
+          os:
+          - debian
+          scope:
+          - functional
+        run:
+          steps:
+          - /usr/bin/v4l2-parser.sh {{ v4l2_driver }}
+      from: inline
+      name: {{ plan }}
+      path: inline/{{ plan }}.yaml
+```
+
+
+```
+cat templates/v4l2-compliance/generic-qemu-v4l2-compliance-template.jinja2 
+```
+
+```
+{% extends 'boot/generic-qemu-boot-template.jinja2' %}
+{% block actions %}
+{{ super () }}
+
+{% include 'v4l2-compliance/v4l2-compliance.jinja2' %}
+
+{% endblock %}
+
+{%- block image_arg %}
+        image_arg: '-kernel {kernel} -append "console={{ console_dev }},115200 root=/dev/ram0 debug verbose {{ extra_kernel_args }}"'
+{%- endblock %}
+```
+
+3. Update `test-plans` section of `test-configs.yaml` with new test definition
+
+
+```yaml
+  v4l2-compliance-vivid:
+    rootfs: debian_buster-v4l2_ramdisk
+    pattern: 'v4l2-compliance/generic-qemu-v4l2-compliance-template.jinja2'
+    params: {v4l2_driver: "vivid"}
+    filters:
+      - whitelist:
+          defconfig:
+            - 'defconfig+virtualvideo'
+            - 'multi_v7_defconfig+virtualvideo'
+```
+
+4. Use `kci_build` to build the kernel and `kci_test` to run the tests in a LAVA lab.
```

Comment:
This is entirely covered in the `kci_build` and `kci_test` pages.  It's fine to add some examples here again as it's convenient to show a full "story", but there should be links and mentions of the other pages for more detailed documentation.